box-content-sdk/src/main/java/com/box/androidsdk/content/views/BezelImageView.java

```diff
@@ -154,16 +154,22 @@ protected void onDraw(Canvas canvas) {
                 mMaskDrawable.draw(cacheCanvas);
                 mMaskedPaint.setColorFilter((mDesaturateOnPress && isPressed())
                         ? mDesaturateColorFilter : null);
-                cacheCanvas.saveLayer(mBoundsF, mMaskedPaint,
-                        Canvas.HAS_ALPHA_LAYER_SAVE_FLAG | Canvas.FULL_COLOR_LAYER_SAVE_FLAG);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                    cacheCanvas.saveLayer(mBoundsF, mMaskedPaint);
+                } else {
+                    cacheCanvas.saveLayer(mBoundsF, mMaskedPaint, Canvas.ALL_SAVE_FLAG);
```

Comment:
Shouldn't there also be the right side of the or expression present ?

Reply:
https://developer.android.com/reference/android/graphics/Canvas.html#saveLayer(android.graphics.RectF,%2520android.graphics.Paint,%2520int)
Isn't this suggesting `cacheCanvas.saveLayer(mBoundsF, mMaskedPaint);` should always be used, because the default is `Canvas.ALL_SAVE_FLAG ` anyways?
Also, why Lollipop, and not Oreo (API level 26)?

Reply:
From linked documentation: 
`cacheCanvas.saveLayer(mBoundsF, mMaskedPaint);` added in API 21 so the check is this case is okay and yes `cacheCanvas.saveLayer(mBoundsF, mMaskedPaint);` by default uses `Canvas.ALL_SAVE_FLAG` internally (If I got the API here right this should point to source https://android.googlesource.com/platform/frameworks/base/+/android-9.0.0_r8/graphics/java/android/graphics/Canvas.java#486)

But, 
looking at latest version (https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/graphics/java/android/graphics/Canvas.java#316) you can see that these constants are still present, thou `@removed`. Here is a good explanation of what ALL_SAVE_FLAG is actually https://stackoverflow.com/a/54201200/2135888 

Since we are targeting minSDK version of 16 and not 21 we should probably target also this case with something like:

```
private static final int CLIP_SAVE_FLAG = 0x02;
    private static final int CLIP_TO_LAYER_SAVE_FLAG = 0x10;
    private static final int MATRIX_SAVE_FLAG = 0x01;
    private static final int SAVE_FLAGS = CLIP_SAVE_FLAG | CLIP_TO_LAYER_SAVE_FLAG | MATRIX_SAVE_FLAG;

    @SuppressLint("WrongConstant")
    private void saveLayerCompat(Canvas canvas, RectF rect, Paint paint, boolean all) {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
            // This method was deprecated in API level 26 and not recommended since 22, but its
            // 2-parameter replacement is only available starting at API level 21.
            canvas.saveLayer(rect, paint, all ? Canvas.ALL_SAVE_FLAG : SAVE_FLAGS);
        } else {
            canvas.saveLayer(rect, paint);
        }
    }
```